celery/backends/base.py

```diff
@@ -667,8 +670,17 @@ def cleanup(self):
         """Backend cleanup."""
 
     def process_cleanup(self):
-        """Cleanup actions to do at the end of a task worker process."""
-
+        try:
+            from eventlet import corolocal
+            task_backend_cache = self.app.local
+            # Whenever an eventlet pool spawns a coroutine it has a different identifier and hence will acquire
+            # a new coroutine-local storage. To avoid that we leak connections we explicitly clean out the
+            # cache once the coroutine is done.
+            if isinstance(task_backend_cache, corolocal.local):
+                task_backend_cache.__dict__.clear()
+        except ImportError:
+            # Eventlet is not installed and hence not used
+            pass
```

Comment:
I'm not sure I am content with coupling eventlet or any other concurrency mode with the backend. Is there another way to resolve this issue?